crates/stages/src/stages/hashing_account.rs

```diff
@@ -171,25 +161,24 @@ impl<DB: Database> Stage<DB> for AccountHashingStage {
         tx: &mut Transaction<'_, DB>,
         input: ExecInput,
     ) -> Result<ExecOutput, StageError> {
-        let stage_progress = input.stage_progress.unwrap_or_default();
-        let previous_stage_progress = input.previous_stage_progress();
-
-        // read account changeset, merge it into one changeset and calculate account hashes.
-        let from_transition = tx.get_block_transition(stage_progress)?;
-        let to_transition = tx.get_block_transition(previous_stage_progress)?;
+        let range = input.next_block_range();
+        if range.is_empty() {
+            return Ok(ExecOutput::done(*range.end()))
+        }
+        let (from_block, to_block) = range.into_inner();
 
         // if there are more blocks then threshold it is faster to go over Plain state and hash all
         // account otherwise take changesets aggregate the sets and apply hashing to
         // AccountHashing table. Also, if we start from genesis, we need to hash from scratch, as
         // genesis accounts are not in changeset.
-        if to_transition - from_transition > self.clean_threshold || stage_progress == 0 {
+        if to_block - from_block > self.clean_threshold || from_block == 1 {
```

Comment:
`from_block == 1` it mens we never executed, it is first iteration. We need to execute `[from,to]` blocks.

It was because the `TransitionId` started from `0`, and this is not true for blocks as we have genesis and we start from block `1`.